include/nbl/system/IFile.h

```diff
@@ -107,4 +108,17 @@ class IFile : public IFileBase, private ISystem::IFutureManipulator
 
 }
 
+namespace nbl::asset {
+	class IFileBackedCPUBuffer : public ICPUBuffer {
+
+		IFileBackedCPUBuffer(core::smart_refctd_ptr<nbl::system::IFile> _file)
+			: ICPUBuffer(_file->getSize(), file->getMappedPointer()), file(_file) {}
```

Comment:
needs a factory `static core::smart_refctd_ptr<IFileBackedCPUBuffer> create` method, because if the file is not mappable (creation flags don't include it) the creation needs to fail and a nullptr needs to be returned

Also you're forwarding to the contructor wrong, just passing the mapped pointer will still cause a copy of the memory to be allocated

You need to refine the constructor, and provide overloads for `getPointer` (both const and non-const), and a bunch of other methods, like convertToDummy etc.

Reply:
Also make the `IFileBackedCPUBuffer` constructor take an offset (default 0) and size, so we can create "views" over sub-sections of the file

Reply:
write this up in a GH issue you'll open about a streaming cache